crates/noirc_evaluator/src/ssa_refactor/ssa_gen/mod.rs

```diff
@@ -37,8 +38,14 @@ pub(crate) fn generate_ssa(program: Program) -> Ssa {
     // to generate SSA for each function used within the program.
     while let Some((src_function_id, dest_id)) = context.pop_next_function_in_queue() {
         let function = &context.program[src_function_id];
-        function_context.new_function(dest_id, function.name.clone(), &function.parameters);
+        function_context.new_function(
+            dest_id,
+            function.name.clone(),
+            &function.parameters,
+            function.unconstrained,
```

Comment:
Maybe we should pass the whole function as a parameter at this point

Reply:
Done